src/Agent/NewRelic/Agent/Core/DataTransport/DataStreamingService.cs

```diff
@@ -692,6 +702,48 @@ private void ExecuteConsumer(PartitionedBlockingCollection<TRequest> collection)
             }
         }
 
+        private bool DequeueItems(PartitionedBlockingCollection<TRequest> collection, int maxBatchSize, CancellationToken cancellationToken, out IList<TRequest> items)
+        {
+            items = null;
+            if(!collection.Take(out var firstItem, cancellationToken))
+            {
+                return false;
+            }
+
+            if(cancellationToken.IsCancellationRequested)
+            {
+                return false;
+            }
+
+            items = new List<TRequest>();
```

Comment:
Should we preallocate this list to the `maxBatchSize` to avoid possible extra allocations and copying?

Reply:
I don't think we should until we know more about how many items are being pulled using this model.   Our queue size is 500, but if things are working well, each of the workers should be pulling far fewer items than that.  So, if we preallocated, we may cause more memory consumption.

* Let's pull a supportability metric to see how many items we are (on average) pulling for each worker.